src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs

```diff
@@ -755,13 +759,27 @@ void BuildDependencies(XmlNodeList dependencies)
                         {
                             if (dependency.NodeType != XmlNodeType.Comment && dependency.NodeType != XmlNodeType.Whitespace)
                             {
+                                DependencyType type;
+                                switch (dependency.ParentNode.Name)
+                                {
+                                    case "ProductDependencies":
+                                        type = DependencyType.Product;
+                                        break;
+                                    case "ToolsetDependencies":
```

Comment:
Might be much since we might not have more dependency types, but we could have a class/struct where the types are defined as constants